contracts/payment/DelegateManager.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract DelegateManager {
+    mapping(address => bool) internal _delegates;
+
+    constructor(address[] memory delegates) {
+        addDelegates(delegates);
+    }
+
+    function addDelegates(address[] memory addresses) public virtual {
```

Comment:
This function does not do any permission management. I see you inherit this class and then call these functions with a permission check. You should mark this class as abstract and make the functions internal, to ensure these functions are never called from the outside.

Reply:
replaced by Openzeppelin's AccessControl